ir/memory.cpp

```diff
@@ -519,8 +535,6 @@ expr Memory::alloc(const expr &size, unsigned align, BlockKind blockKind,
   Pointer p(*this, bid, is_local);
   auto short_bid = p.get_short_bid();
   // TODO: If address space is not 0, the address can be 0.
-  // TODO: address of new blocks should be disjoint from other live blocks
-  // FIXME: non-zero constraint can go away when disjoint constr is in place
```

Comment:
this last comment wasn't addressed.
It's unclear we want to keep the non-null constraint. Maybe yes, but needs to be evaluated.

Reply:
Null pointer block has size 0, so the non-null condition is still needed. It allows two blocks to be adjacent.

Reply:
adjacent, yes, but not at address zero?

Reply:
never mind, you're right.